### PR TITLE
Add trivy scan to docker images of quick deployable samples

### PR DESCRIPTION
**Pull request by G5andeepD**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: .azure/pr-check.yaml [hunk 1]
```diff
@@ -23,5 +30,39 @@ steps:
...
+
+  - template: install/install-trivy.yaml@common-templates
+  - script: |
+        echo "##[group]Running Trivy scan for each image"
+
+        # Initialize exit_code as 0 (success)
+        exit_code=0
+
+        # Read image_urls.txt and pass each image to trivy-scan.sh
+        while IFS= read -r image; do
```
Comment (chameerar) [inline .azure/pr-check.yaml:47]: Do we need `IFS= ` here? Are we expecting whitespace in the image URLs?

File: .azure/pr-check.yaml [hunk 2]
```diff
@@ -23,5 +30,39 @@ steps:
...
     displayName: "Metadata Validation"
 
+  # Give execute permissions to trivy-scan.sh
+  - script: |
+      echo "Giving execute permission to trivy-scan.sh"
+      chmod +x .azure/scripts/trivy-scan.sh
+    displayName: "Set execute permissions"
+
+  - template: install/install-trivy.yaml@common-templates
+  - script: |
```
Comment (chameerar) [inline .azure/pr-check.yaml:40]: Shall we remove the comments where the code is obvious? The code will look cleaner that way.

File: .azure/pr-check.yaml [hunk 3]
```diff
@@ -23,5 +30,39 @@ steps:
...
+            echo "Trivy scan failed for image: $image"
+            exit_code=1  # Mark exit_code as failure
+          fi
+        done < image_urls.txt
+
+        echo "##[endgroup]"
+
+        # Exit with the appropriate exit code to fail the step if any scan failed
+        exit $exit_code
+    displayName: "Run Trivy scan for all images"
```
Comment (chameerar) [inline .azure/pr-check.yaml:65]: I think it's better to say
```suggestion
    displayName: "Run Trivy scan for quick deployable images"
```
WDYT?

**Review by chameerar — CHANGES_REQUESTED** [review]

File: .azure/pr-check.yaml (hunk 3 above)
Comment (G5andeepD) [reply .azure/pr-check.yaml:65]: +1

File: .azure/pr-check.yaml (hunk 1 above)
Comment (G5andeepD) [reply .azure/pr-check.yaml:47]: Image urls dont have whitespaces so we can remove IFS.

File: .azure/build.yaml [hunk 4]
```diff
@@ -36,6 +36,43 @@ steps:
...
+    displayName: "Run Trivy scan for quick deployable images"
+
+  - script: |
+        echo "##[group]Pushing images to ACR"
+        while read -r image; do
+          echo "Pushing image: $image"
+          docker push "$image"
+        done < image_urls.txt
+        echo "##[endgroup]"
+    displayName: 'Push Docker Images'
```
Comment (chameerar) [inline .azure/build.yaml:74]:
```suggestion
    displayName: 'Push Quick Deployable Docker Images'
```

**Review by chameerar — CHANGES_REQUESTED** [review]

**Review by chameerar — APPROVED** [review]
LTGM

**Pull request merged**